RecoJets/JetPlusTracks/plugins/JetPlusTrackCorrector.cc

```diff
@@ -1086,7 +1294,34 @@ bool JetPlusTrackCorrector::matchMuons(TrackRefs::const_iterator& itrk, const ed
       return false;
     }
 
-    if (*itrk == imuon->innerTrack()) {
+    if (*itrk == imuon->innerTrack())
+      return true;
+  }
+
+  return false;
+}
+
+bool JetPlusTrackCorrector::matchMuons(TrackRefs::const_iterator& itrk,
+                                       const edm::Handle<pat::MuonCollection>& muons) const {
+  if (muons->empty()) {
+    return false;
+  }
+
+  pat::MuonCollection::const_iterator imuon = muons->begin();
+  pat::MuonCollection::const_iterator jmuon = muons->end();
+
+  for (; imuon != jmuon; ++imuon) {
+    if (imuon->innerTrack().isNull() == 1)
+      continue;
+    //  std::cout<<"JetPlusTrackCorrector::matchMuons::Track "<<(**itrk).pt()<<" "
+    //  <<(**itrk).eta()<<" "<<(**itrk).phi()<<" Muon "<<
+    // (*(imuon->innerTrack())).pt()<<" "<< (*(imuon->innerTrack())).eta()<<" "<<
+    // (*(imuon->innerTrack())).phi()<<std::endl;
+    if (fabs((**itrk).pt() - (*(imuon->innerTrack())).pt()) < 0.1 &&
```

Comment:
I'm wondering about the origin of the pt, eta and phi cutoffs that determine the identity of a track and a muon. I could suggest to put them as a configuration flag, very explicitly, rather than hardcoding here, as it may go unnoticed in the future.

Reply:
done